### PR TITLE
Remove access to contiguous TL buffer from Coco Reader tests

### PR DESCRIPTION
**Pull request by klecki**
## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [x] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [ ] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
Remove access to contiguous TL buffer from Coco Reader tests.
Introduce sample-by-sample copies.

#### Additional information
- Affected modules and functionalities:
Coco reader tests

- Key points relevant for the review: 

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [x] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: dali/operators/reader/coco_reader_op_test.cc [hunk 1]
```diff
@@ -112,10 +112,15 @@ class CocoReaderTest : public ::testing::Test {
...
-      shape.size() * sizeof(int));
+    auto *out_ptr = ids.data();
+    for (int sample_idx = 0; sample_idx < shape.num_samples(); sample_idx++) {
+      MemCopy(
+        out_ptr,
+        output.tensor<int>(sample_idx),
+        shape[sample_idx].num_elements() * sizeof(int));
+      out_ptr += shape[sample_idx].num_elements();
+    }
+    cudaStreamSynchronize(0);
```
Comment (JanuszL) [inline dali/operators/reader/coco_reader_op_test.cc:123]: You don't need that. You are using pagable memory in the copy so it syncs by definition.

File: dali/operators/reader/coco_reader_op_test.cc [hunk 2]
```diff
@@ -197,10 +202,15 @@ class CocoReaderTest : public ::testing::Test {
...
-      obj_count * bbox_size_ * sizeof(float));
+    auto *boxes_ptr = boxes.data();
+    for (int sample_idx = 0; sample_idx < boxes_shape.num_samples(); sample_idx++) {
+      MemCopy(
+        boxes_ptr,
+        boxes_output.tensor<float>(sample_idx),
+        boxes_shape[sample_idx].num_elements() * sizeof(float));
+      boxes_ptr += boxes_shape[sample_idx].num_elements();
+    }
+    cudaStreamSynchronize(0);
```
Comment (JanuszL) [inline dali/operators/reader/coco_reader_op_test.cc:213]: As above

File: dali/operators/reader/coco_reader_op_test.cc [hunk 3]
```diff
@@ -229,10 +239,15 @@ class CocoReaderTest : public ::testing::Test {
...
-      obj_count * sizeof(int));
+    auto *labels_ptr = labels.data();
+    for (int sample_idx = 0; sample_idx < labels_shape.num_samples(); sample_idx++) {
+      MemCopy(
+        labels_ptr,
+        labels_output.tensor<int>(sample_idx),
+        labels_shape[sample_idx].num_elements() * sizeof(int));
+      labels_ptr += labels_shape[sample_idx].num_elements();
+    }
+    cudaStreamSynchronize(0);
```
Comment (JanuszL) [inline dali/operators/reader/coco_reader_op_test.cc:250]: Again

File: dali/operators/reader/coco_reader_op_test.cc [hunk 4]
```diff
@@ -257,15 +272,24 @@ class CocoReaderTest : public ::testing::Test {
...
-        vertices_gt_.size() * sizeof(float));
+      auto *vertices_ptr = vertices.data();
+      for (int sample_idx = 0; sample_idx < vertices_shape.num_samples(); sample_idx++) {
+        MemCopy(
+          vertices_ptr,
+          vertices_output.tensor<float>(sample_idx),
+          vertices_shape[sample_idx].num_elements() * sizeof(float));
+        vertices_ptr += vertices_shape[sample_idx].num_elements();
+      }
+      cudaStreamSynchronize(0);
```
Comment (JanuszL) [inline dali/operators/reader/coco_reader_op_test.cc:292]: Same

**Review by JanuszL — APPROVED** [review]

**Review by stiepan — APPROVED** [review]

**Pull request merged**